### ISSUE TITLE
Using QueryCollection fetch shorthand without params causes error

### ISSUE THREAD

**Issue by j4m3s** [PhpGt/Database#35]
```
QueryCollection->query("something");  // works
QueryColelction->something(["binky" => "boo"]); // works
QueryCollection->something();


There was 1 error:

1) Gt\Database\Query\QueryCollectionTest::testQueryShorthand
Undefined offset: 0
```

**Comment by j4m3s** [comment]
This is a lesson in the limitations of code coverage stats (in this case combined with magic methods).  Code coverage was showing:
```
\Gt\Database\Query::QueryCollection
  Methods: 100.00% ( 4/ 4)   Lines: 100.00% ( 13/ 13)
```
but the shorthand notation wasn't actually being tested when not receiving any params.  There is no alternative but to use magic methods to implement the shorthand notation, but the code coverage stats gave a false sense of security!

**Comment by g105b** [comment]
I've made mistakes from the false sense of security of 100% coverage before. Good catch.

**Comment by j4m3s** [comment]
Closed by PR #36
